server/dive_tasks/tasks.py

```diff
@@ -545,3 +545,71 @@ def convert_images(self: Task, folderId):
+    manager: JobManager = patch_manager(self.job_manager)
+    if utils.check_canceled(self, context):
+        manager.updateStatus(JobStatus.CANCELED)
+        return
+
+    with tempfile.TemporaryDirectory() as _working_directory, suppress(utils.CanceledError):
+        _working_directory_path = Path(_working_directory)
+        item: GirderModel = gc.getItem(itemId)
+        file_name = str(_working_directory_path / item['name'])
+        manager.write(f'Fetching input from {itemId} to {file_name}...\n')
+        gc.downloadItem(itemId, _working_directory, item["name"])
+        with zipfile.ZipFile(file_name, 'r') as zipObj:
+            listOfFileNames = zipObj.namelist()
+            sum_file_size = sum([data.file_size for data in zipObj.filelist])
+            sum_compress_size = sum([data.compress_size for data in zipObj.filelist])
+            ratio = sum_file_size / sum_compress_size
+            if ratio > 600:
+                manager.write(
+                    f"Compression ratio is exceedingly high at {ratio}\n\
+                    Please contact an admin at viame-web@kitware.com if this is a valid zip file"
+                )
+                manager.updateStatus(JobStatus.ERROR)
+                return
+
+            for fileName in listOfFileNames:
+                if fileName.endswith('.zip'):
+                    manager.write("Nested Zip files are invalid\n")
+                    manager.updateStatus(JobStatus.ERROR)
+                    return
+                manager.write(f"Extracting: {fileName}\n")
+                zipObj.extract(fileName, f'{_working_directory}/{fileName}')
+        # validation of files in folder using dive/data
+        validation = gc.sendRestRequest(
+            'POST', '/dive_dataset/validate_files', json=listOfFileNames
+        )
+        if validation.get('ok', False):
+            manager.write(f"Annotations: {validation['annotations']}\n")
+            manager.write(f"Media: {validation['media']}\n")
+            dataset_type = validation['type']
+            manager.write(f"Type: {dataset_type}\n")
+            # Upload all resulting items back into the root folder
+            manager.updateStatus(JobStatus.PUSHING_OUTPUT)
+            gc.upload(f'{_working_directory}/**/*', folderId)
+            gc.addMetadataToFolder(str(folderId), {constants.TypeMarker: dataset_type})
+            gc.addMetadataToItem(str(itemId), {constants.ZipFileExtractedMarker: True})
+            # create a source folder to place the zipFile inside of
+            metadata = {constants.ZipFileExtractedMarker: True}
+            created_folder = gc.createFolder(
+                folderId,
+                constants.SourceFolderName,
+                reuseExisting=True,
+            )
+            gc.sendRestRequest(
+                "PUT",
+                f"/item/{str(itemId)}?name={item['name']}&folderId={str(created_folder['_id'])}",
+                json=metadata,
+            )
```

Comment:
I think this PUT and the metadata update should happen just before the directory upload instead of after it.

As is, you could end up in a loop where stuff keeps getting uploaded, but the zip never moves or has its metadata updated, if, for example, an exception is encountered during upload.   In a worst case, this could just run over and over until disk fills up.

Then, if there's a problem, the celery task will report that it failed and it won't be able to run again.